docs/proposals/PROPOSAL_G.md

```diff
@@ -0,0 +1,63 @@
+# Proposal G - Image Compatibility Artifact
+
+The proposal introduces a new `application/vnd.oci.image-compatibility.v1` artifact type.
+Interest groups/communities SHOULD implement tools to address their specific (or common) use cases.
+
+## Artifact Manifest
+
+An artifact manifest MAY contain multiple blobs with different or the same media types, but MUST contain at least one.
+The new artifact type for image compatibility MUST be referenced in OCI documentation.
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.oci.image-compatibility.v1",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2
+  },
+  "layers": [
+    {
+      "mediaType": "application/k8s.nfd.image-compatibility.spec.v1alpha1+json",
+      "digest": "sha256:4a47f8ae4c713906618413cb9795824d09eeadf948729e213a1ba11a1e31d052",
+      "size": 1710,
+      "annotations": {
+        "name": "profileA"
+      }
+    }, 
+    {
+      "mediaType": "application/k8s.nfd.image-compatibility.spec.v1alpha1+json",
+      "digest": "sha256:35589b7dcd49cbcc48a22815aceccaf34190189a3f25c6e84772771e051a992c",
+      "size": 450,
+      "annotations": {
+        "name": "profileB"
+      }
+    },
+    {
+      "mediaType": "application/k8s.nfd.image-compatibility.spec.v1alpha1+json",
+      "digest": "sha256:494b04517cada9ba77ebdd3a94ad3e0e10a4d000dada8cb0876cd3c5e90fc826",
+      "size": 2040,
+      "annotations": {
+        "name": "profileC"
+      }
+    }
+  ],
+  "annotations": {
+    "oci.opencontainers.image.created": "2024-01-02T03:04:05Z"
+  }
+}
+```
+
+## Compatibility Specification
+
+Interest groups/communities are responsible to release their own media types with corresponding schemas and overall JSON structure.
+Schema MUST provide attributes that will allow to express image compatibility.
+Known media types SHOULD be referenced in the OCI documentation.
+
+## Tools implementation
+
+Tools that implement image compatibility use cases MUST follow the schema provided by the media type.
```

Comment:
In practice this means a 1:1 mapping of tools to media types... :/

Reply:
Only when we talk about attributes. IMO we should avoid adding more to spec than a media type allows. That would guarantee to have the same expectations from spec side across different implementations. If it's required to add something extra (like schema graphs, compatibility relations) we should use annotations.

Reply:
But if there is no definition of the structure, you could have entirely different json blobs and they would both be valid, but (in real world cases) only supported by whatever tooling the community provides. If a community makes  a nice validation tool for the structure, nobody else can use it except for them, for example.

Reply:
Definition of the structure comes from the media type. This is the same as for all OCI media types.
https://github.com/opencontainers/image-spec/blob/main/media-types.md
All the allowed properties are tight to specifc type.
IMO it would be wrong to keep the same structure across different media types.

Reply:
We can have one media type for the compatibility spec and add information about source (like k8s, nfd etc.) over media type parameters.

That way we provide one spec under one media type across different compatibility keys providers.
That could look like this
```
"mediaType": "application/vnd.oci.image-compatibility.spec.v1+json; labels=k8s-nfd" # only k8s nfd labels
"mediaType": "application/vnd.oci.image-compatibility.spec.v1+json; labels=hpc" # only hpc labels
"mediaType": "application/vnd.oci.image-compatibility.spec.v1+json; labels=k8s-nfd,hpc" # combined labels
```
(We don't have to combine labels, but I guess the general idea is clear. Also we can name the parameter differently.)